feat(docker): build centreon-web docker image

kduret opened the pull request:
## Description

feat(docker): build centreon-web docker image

THIS PR IS FOR TESTING PURPOSE BY @Tpo76

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

kduret committed:
feat(docker): build centreon-web docker image

Rebase

Tpo76 committed:
First generation test

Tpo76 committed:
Trying Again

Tpo76 committed:
Remove most of the centreon-web dependancies

Tpo76 committed:
New deb dependencies for centreon-web docker php-fpm only

Tpo76 committed:
remove comment

kduret committed:
align centreon web workflow

Tpo76 committed:
1

Tpo76 committed:
Add back repo in image

Tpo76 committed:
fixing VERSION_CODE gpg dependencies

Tpo76 committed:
update

Tpo76 committed:
Fixing error The repository 'https://packages.centreon.com/apt-standard-25.05-stable bookworm Release' does not have a Release file.

Tpo76 committed:
Fixing error The repository 'https://packages.centreon.com/apt-standard-25.05-stable bookworm Release' does not have a Release file.

Tpo76 committed:
Fix

Tpo76 committed:
Fix

Tpo76 committed:
Remove all deps

Tpo76 committed:
Fix shared link

Tpo76 committed:
Fix shared link

Tpo76 committed:
Add back centreon-common dependency to centreon-web

Tpo76 committed:
Add back centreon-common dependency to centreon-web

Tpo76 committed:
Fix symfony stuff

Tpo76 committed:
Fix symfony stuff

Tpo76 committed:
Fix few directory dependency

Tpo76 committed:
Fix few directory dependency

Tpo76 committed:
.

Tpo76 committed:
.

Tpo76 committed:
Commit don't want to sync up

Tpo76 committed:
Commit don't want to sync up

Tpo76 committed:
Fixing typo in Dockerfile

Tpo76 committed:
Fixing typo in Dockerfile

Tpo76 committed:
fix permission and path

Tpo76 committed:
fix permission and path

Tpo76 committed:
Fixing basic installation process issue

Tpo76 committed:
Fixing basic installation process issue

Tpo76 committed:
adding user/group management + centreon-perl-libs deps

Tpo76 committed:
adding user/group management + centreon-perl-libs deps

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Tpo76 committed:
fix dependency error

Tpo76 committed:
fix dependency error

Tpo76 committed:
Merge branch 'fth-docker-centreon-web' of https://github.com/centreon/centreon into fth-docker-centreon-web

Tpo76 committed:
sync dev derivations/centreon-web

Tpo76 committed:
sync dev derivations/centreon-web

kduret committed:
fix(docker): fix build of centreon-web image

kduret committed:
fix(docker): fix build of centreon-web image

Tpo76 committed:
Remove centreon-perl-libs installing to much sub dependency

Tpo76 committed:
Remove centreon-perl-libs installing to much sub dependency

Tpo76 committed:
Exclude centreon-common for subdependencies

Tpo76 committed:
Exclude centreon-common for subdependencies

Tpo76 committed:
Add missing directory provide by centreon-common

Tpo76 committed:
Add missing directory provide by centreon-common

Tpo76 committed:
Add directory comming from centreon-common

Tpo76 committed:
Add directory comming from centreon-common

Tpo76 committed:
Auto-install first try

Tpo76 committed:
Auto-install first try

Tpo76 committed:
Adjust auto-intall

Tpo76 committed:
Adjust auto-intall

Tpo76 committed:
Add default CMD

Tpo76 committed:
Add default CMD

Tpo76 committed:
Add custom SQL container configuration for broker-module

Tpo76 committed:
Add custom SQL container configuration for broker-module

Tpo76 committed:
Fix install file

Tpo76 committed:
Fix install file

Tpo76 committed:
Add www-data to centreon-broker and centreon-engine for config generattion

Tpo76 committed:
Add www-data to centreon-broker and centreon-engine for config generattion

Tpo76 committed:
extend user/group permission

Tpo76 committed:
extend user/group permission

Tpo76 committed:
fix user/group

Tpo76 committed:
fix user/group

Tpo76 committed:
Add DB permissions

Tpo76 committed:
Add docker-compose file and Dockerfile

Tpo76 committed:
Add docker-compose file and Dockerfile

Tpo76 committed:
update compose collect & sudo permission for centreon-web

Tpo76 committed:
update compose collect & sudo permission for centreon-web

Tpo76 committed:
manage sudoer and "service" script

Tpo76 committed:
manage sudoer and "service" script

pkippes committed:
chore(release) bump web 25.05.1

pkippes committed:
chore(release) bump web 25.05.1

pkippes committed:
chore(release) Create Update-25.05.1.php

Undo orgin/master merge

pkippes committed:
chore(release) Create Update-25.05.1.php

Undo orgin/master merge

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Tpo76 committed:
Various changes

Tpo76 committed:
Various changes

Tpo76 committed:
Fix cbmod path

Tpo76 committed:
Fix cbmod path

Tpo76 committed:
Adjustement on centreon gorgone right and permissions to reload engine

Tpo76 committed:
Adjustement on centreon gorgone right and permissions to reload engine

Tpo76 committed:
Add default gorgone config file + gorgone_api_address

Tpo76 committed:
Add default gorgone config file + gorgone_api_address

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Tpo76 committed:
Try to fix job error (maybe bad idea)

Tpo76 committed:
Try to fix job error (maybe bad idea)

Tpo76 committed:
Undo workflows modification

Tpo76 committed:
Undo workflows modification

pkippes committed:
chore(release) bump web 25.05.2

pkippes committed:
chore(release) bump web 25.05.2

kduret committed:
Merge branch 'develop' into fth-docker-centreon-web

kduret committed:
fix(ci): do not send empty translations to frontend (#8095)

technique-ci committed:
chore(lang): update translations

kduret committed:
fix(packaging): fix dsm purge file permissions (#7882)

itoussies committed:
fix(cron): fix centreon-token log file permissions (#7768)

Co-authored-by: Kevin Duret <kduret@centreon.com>

asalhi-ext committed:
fix(META SERVICE): fix ACL on adding a Meta Service (#8097)

adr-mo committed:
feat(acl): implement ACL on image folders (#7923)

Co-authored-by: Jérémy Delpierre <jdelpierre@users.noreply.github.com>
Co-authored-by: dmyios <diosypenko@centreon.com>
Co-authored-by: Dmytro Iosypenko <108675430+dmyios@users.noreply.github.com>
Co-authored-by: Anthony Graule <agraule@centreon.com>

adr-mo committed:
feat(acl): implement ACL on image folders (#7923)

Co-authored-by: Jérémy Delpierre <jdelpierre@users.noreply.github.com>
Co-authored-by: dmyios <diosypenko@centreon.com>
Co-authored-by: Dmytro Iosypenko <108675430+dmyios@users.noreply.github.com>
Co-authored-by: Anthony Graule <agraule@centreon.com>

technique-ci committed:
chore(lang): update translations

technique-ci committed:
chore(lang): update translations

technique-ci committed:
chore(deps): absorb 2025-08 dependabot GitHub Actions updates (#7943)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

a-launois committed:
fix(conf) fix update macro services with vault (#8038)

Co-authored-by: selfakiri <selfakiri@ext.centreon.com>

marlaud-centreon committed:
feat(resource-configuration): migrate create service category endpoint (#7819)

asalhi-ext committed:
fix(OPEN TICKET): enhance listing of opentickets rules (#8074)

selfakiri committed:
fix(services): in cloud a command should be required for services (#8079)

selfakiri committed:
fix(configuration): fix error msg in export (#8089)

a-launois committed:
fix(conf) fix alias field in form (#8076)

dmyios committed:
feat(Dashboards): Hostgroups availability widget (#7827)

Co-authored-by: Tom Darneix <tomdar87@outlook.com>
Co-authored-by: Adrien Morais-Mestre <31647811+adr-mo@users.noreply.github.com>
Co-authored-by: Adrien Morais <amorais@centreon.com>
Co-authored-by: Nouha-ElAbrouki <97687698+Noha-ElAbrouki@users.noreply.github.com>
Co-authored-by: nouha <nelabrouki-ext@centreo.com>
Co-authored-by: smau <102975713+smau-centreon@users.noreply.github.com>

dmyios committed:
feat(Dashboards): Hostgroups availability widget (#7827)

Co-authored-by: Tom Darneix <tomdar87@outlook.com>
Co-authored-by: Adrien Morais-Mestre <31647811+adr-mo@users.noreply.github.com>
Co-authored-by: Adrien Morais <amorais@centreon.com>
Co-authored-by: Nouha-ElAbrouki <97687698+Noha-ElAbrouki@users.noreply.github.com>
Co-authored-by: nouha <nelabrouki-ext@centreo.com>
Co-authored-by: smau <102975713+smau-centreon@users.noreply.github.com>

technique-ci committed:
chore(lang): update translations

technique-ci committed:
chore(lang): update translations

selfakiri committed:
fix(ldap): contacts with special chars shouldn't be imported more than once (#8094)

MedWassCh committed:
chore(Test Perf) : Adding performance testing (#6661)

mushroomempires committed:
fix(ci): replace jf_url from jfrog.io to packages.centreon.com (#8132)

selfakiri committed:
fix(logs): show special chars in logs details (#8098)

Commit:
Fix the flacky test 13-uri-configuration on nightly (#8138)

tuntoja committed:
fix(unattended): fix mariadb secure setup (#8122)

mushroomempires committed:
fix(delivery-feature): fix label condition on delivery_type check in get-environment (#8145)

mushroomempires committed:
fix(nightly-ci): add back condition 'starts with dev' on branch for nightly ticket creation

Drokha committed:
Fix(Dashboard): playlist rotation fail not found after dashboard edit (#7701)

Co-authored-by: Tom Rabut <trabut@mac.home>
Co-authored-by: MedWassCh <wchaabane@centreon.com>

kduret committed:
feat(ci): allow parallel delivery of release and hotfix (#7924)

kduret committed:
feat(ci): allow parallel delivery of release and hotfix (#7924)

tuntoja committed:
fix(ci): update references to bitnami mariadb 10.11 in dockerfiles and workflows (#8150)

Tpo76 committed:
new gorgone Dockerfile and configuration web adjustement and various userpermission

Tpo76 committed:
new gorgone Dockerfile and configuration web adjustement and various userpermission

Tpo76 committed:
Resolve conflicts

Tpo76 committed:
Resolve conflicts

Tpo76 committed:
Merge remote-tracking branch 'origin/develop' into fth-docker-centreon-web

Tpo76 committed:
fix conflict after rebase

Tpo76 committed:
fix conflict after rebase

Tpo76 committed:
update health check for compose

Tpo76 committed:
update health check for compose

Tpo76 committed:
Merge branch 'fth-docker-centreon-web' of https://github.com/centreon/centreon into fth-docker-centreon-web

Tpo76 committed:
del docker-centreon-web workflow

Tpo76 committed:
add back docker workflow

Tpo76 committed:
Fix help.po after rebase

Tpo76 committed:
Fix help.po part 2

Tpo76 committed:
keep trying to fix workflow

Tpo76 committed:
Hardcode for '_CENTREON_PATH_'

Tpo76 committed:
Keep pushing

Tpo76 committed:
undone changes

kduret committed:
Merge branch 'develop' into fth-docker-centreon-web

kduret committed:
retrieve changes from develop

kduret committed:
retrieve changes from develop

kduret committed:
retrieve changes from develop

Tpo76 committed:
Troubleshooting build error

Tpo76 committed:
Change user ID to sysuser force from image to boowkworm

Tpo76 committed:
Fix installation process and repo

Tpo76 committed:
Fix installation script and permissions

Tpo76 committed:
Fix installed insert

Tpo76 committed:
Add and manage engine-context.json file

Tpo76 committed:
Fix directory issue Centreon web/ include default gorgone 40-gorgoned.yaml

Tpo76 committed:
Adjust service to systemctl for compatibility with Debian 12

Tpo76 committed:
Fix dockerfiles issues

Tpo76 committed:
Enhance gorgone deployment to support poller and central configuration

Tpo76 committed:
Enhanced gorgone image to support both central and poller modes.

Tpo76 committed:
Add dummy centengine file to avoid errors

Tpo76 committed:
Workaround centengine debug

Tpo76 committed:
fix typo

Tpo76 committed:
Fine tune various components permissions

Tpo76 committed:
Auto export config at start

Tpo76 committed:
workaround root file creation when using clapi to export configuration

Tpo76 committed:
fix when php pod recreate

Tpo76 committed:
Adjust permissions and export configuration

Tpo76 committed:
Update volume permisssions for centcache

Tpo76 committed:
Modify rrd output to use only 1 broker service

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Tpo76 committed:
fix cencore.cmd permission + various improvements

Tpo76 committed:
Merge branch 'fth-docker-centreon-web' of https://github.com/centreon/centreon into fth-docker-centreon-web

Tpo76 committed:
Redo gorgone docker for poller plus poller compose file

Tpo76 committed:
Add env.local support for centreon-web
Add cronbox image with all scripts

Tpo76 committed:
Fixed environment variable handling for APP_ENV

Tpo76 committed:
Extra fixing APP_ENV variable in Dockerfile and docker-compose.yml

Tpo76 committed:
fix env default issue

Tpo76 committed:
modify .env scripts

Tpo76 committed:
Add TZ management

Tpo76 committed:
Fix TZ setting

Tpo76 committed:
Fix timezone issues in Centreon Web container

Tpo76 committed:
Remove "set -e" from timezone script

Tpo76 committed:
Add TZ in php-cli

Tpo76 committed:
Change php-fpm user to www-data

Tpo76 committed:
Remove su commands and use USER directive in Dockerfile instead

Tpo76 committed:
Using gosu to change user in entrypoint script

Tpo76 committed:
Remove gosu user php-fom consider best practice to run master as root

Tpo76 committed:
Make app data dynamic and fix permissions

Tpo76 committed:
Change permission order to avoid issues with files created by www-data user

Tpo76 committed:
Reordering of commands and cleanup of apt cache

Tpo76 committed:
Optimize centreon-web image size managing env variables FOR MYSQL

Tpo76 committed:
Change centcore permissions setup order to avoid permission issues

Tpo76 committed:
Manage Centreon upgrade

Tpo76 committed:
Missing jq dependency in Dockerfile

Tpo76 committed:
Change installation logic

Tpo76 committed:
Fix typos

Tpo76 committed:
Fix mysql command line to use MYSQL_PWD variable

Tpo76 committed:
Create environment variable for configuration

Tpo76 committed:
Fix envsubst

Tpo76 committed:
Adjust database setting

Tpo76 committed:
Fix DB user creation and rights in entrypoint script

Tpo76 committed:
Update centreon/docker/docker-compose.yml with CI centengine image

Tpo76 committed:
Update compose file image

Tpo76 committed:
Optimize centreon-web container: reduce image size, improve security and startup performance

Tpo76 committed:
Resolve .github/workflows/web.yml conflict

Tpo76 committed:
Update gorgone image to use centreon-collect official container

Tpo76 committed:
Improving local testing

Tpo76 committed:
Image size optimizations

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Tpo76 committed:
Fix backend dependancies

Tpo76 committed:
Merge branch 'fth-docker-centreon-web' of https://github.com/centreon/centreon into fth-docker-centreon-web

Tpo76 committed:
Fix backend dependencies installation for web container

Tpo76 committed:
Change permission order in web container setup

kduret committed:
fix

Tpo76 committed:
Fix gpg issue in Dockerfile

Tpo76 committed:
Merge branch 'fth-docker-centreon-web' of https://github.com/centreon/centreon into fth-docker-centreon-web

Tpo76 committed:
Fix /tmp issue

Tpo76 committed:
Fix permissions on generated config directories

Tpo76 committed:
Make export conditional

Tpo76 committed:
fix file permissions

Tpo76 committed:
Add ETCD logic for configuration file management in Centreon Web

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Tpo76 committed:
Fix centreon-web deps after develop merge

Tpo76 committed:
fix etcd process

Tpo76 committed:
Up-to-date docker-compose.yaml

Tpo76 committed:
Fix to not use centcore.cmd file

Tpo76 committed:
Revert "Fix to not use centcore.cmd file"

This reverts commit 22f8b2361669c7f994d1e16d41e16addda8470e7.

Tpo76 commented:
🤖 *This is an automatic comment generated by an internal tool.*

### Complexity Classification: **Standard**

**Justification:** Automatic classification failed; defaulting to standard.

| Metric | Value |
|---|---|
| WIP Points | 2 |
| SLA Track | Standard SLA (2 work days) |
| First review SLA | 1 work day |

A maintainer can override this classification at any time.

Tpo76 commented:
🤖 *This is an automatic comment generated by an internal tool.*

This PR does not appear to be linked to a Jira ticket.

PRs without a linked Jira ticket will **not be prioritized** for review. To get this PR reviewed, please:
1. Create or find the corresponding Jira ticket
2. Add the ticket key (e.g., `MON-12345`) to the PR title or description

Once linked, this PR will be automatically classified and tracked.

Tpo76 committed:
init trap container

Tpo76 committed:
Merge branch 'develop' into fth-docker-centreon-web

Keep new bookworm Docker structure: intentional script deletions
and restructured entrypoint scripts take precedence over develop changes.

Tpo76 committed:
chore(docker): rename bookworm to bookworm-tmp for new web Docker setup

Isolate new bookworm Docker work in bookworm-tmp/ to avoid future
conflicts with the existing bookworm/ setup maintained on develop.

Tpo76 committed:
feat(ci): add dedicated workflow for bookworm-tmp Docker dependencies

New workflow only triggers on bookworm-tmp/ path changes, avoiding
conflicts with existing docker-web-dependencies workflow.
Builds centreon-web-dependencies-bookworm-tmp images independently.

Tpo76 committed:
fix(ci): use bookworm-tmp in docker-centreon-web dockerize matrix

Replace bookworm with bookworm-tmp in the dockerize job matrix to match
the renamed Docker directory. distrib output stays bookworm for cache key
compatibility with the package job (still building Debian bookworm debs).

Tpo76 committed:
chore(docker): restore bookworm/ from develop for coexistence with bookworm-tmp

Both bookworm/ (develop original, untouched) and bookworm-tmp/ (new work)
now coexist. CI on this branch builds bookworm-tmp/ only.

Tpo76 committed:
fix(ci): fix empty tag for packaging-nfpm-bookworm image in docker-trapd

packaging_img_version does not exist in get-environment outputs.
Use major_version instead, consistent with other packaging workflows.

Tpo76 committed:
fix(ci): remove unnecessary token for centreon-collect checkout in docker-trapd

centreon/centreon-collect is a public repo, no token required.
GITHUB_CENTREON_BOT_TOKEN was causing 'Input required and not supplied' error.

Tpo76 committed:
fix(ci): fix cache key distrib_name in docker-trapd dockerize job

parse-distrib action has no distrib_name output, causing empty segment
in cache key (deb--trap). Use matrix.operating_system instead to match
keys written by package jobs (deb-bookworm-trap / deb-bookworm-perl).

Tpo76 committed:
include perl deps

Tpo76 committed:
auto restart with inotifywait

Tpo76 committed:
include entrypoint script

Tpo76 committed:
redirect log
